providers/implementations/keymgmt/ec_kmgmt.c

```diff
@@ -610,6 +687,87 @@ int ec_set_params(void *key, const OSSL_PARAM params[])
     return ec_key_otherparams_fromdata(eck, params);
 }
 
+#ifndef OPENSSL_NO_SM2
+static
+int sm2_get_params(void *key, OSSL_PARAM params[])
+{
+    int ret;
+    EC_KEY *eck = key;
+    const EC_GROUP *ecg = NULL;
+    OSSL_PARAM *p;
+    unsigned char *pub_key = NULL;
+
+    ecg = EC_KEY_get0_group(eck);
+    if (ecg == NULL)
+        return 0;
+
+    if ((p = OSSL_PARAM_locate(params, OSSL_PKEY_PARAM_MAX_SIZE)) != NULL
+        && !OSSL_PARAM_set_int(p, ECDSA_size(eck)))
+        return 0;
+    if ((p = OSSL_PARAM_locate(params, OSSL_PKEY_PARAM_BITS)) != NULL
+        && !OSSL_PARAM_set_int(p, EC_GROUP_order_bits(ecg)))
+        return 0;
+
+    /* XXX:
+     * I dropped the support of OSSL_PKEY_PARAM_SECURITY_BITS since
+     * I didn't find definition of SM2 security bits so far. This could
+     * be supported if the definition is clear in the future.
```

Comment:
Should we assume the same as for ECDH/ECDSA?

Reply:
Not sure about this. Is this a 'MUST' parameter?

Reply:
I would say it is required otherwise it won't work with seclevel/authlevel settings in TLS ad X.509 - if you actually care about SM2 for TLS and X.509 certificates.

Reply:
SM2 for TLS is not needed to consider at current stage. But X.509 would be a problem. Any way that can resolve this issue?

Reply:
If it's a *required* parameter, then I think we need to think about @mattcaswell 's suggestion...

Reply:
For all intents and purposes, an SM2 key is an EC key, with just two differences:

1.  Key generation group order is chosen within an ever so slightly different range.
2.  Signature requires an additional distinguishing ID, which is essentially used to scramble the signature a bit more.

Are these differences enough to make the security bits in an SM2 key different from the security bits in an EC key?  If not, then we should simply return the same for an SM2 key as for an EC key.  *Do note that this is what we have done historically*, so it would also be backward compatible with ourselves.

Also, since we have no other data to lean on for the moment, I don't think we have much other choice than to lean on already existing knowledge, and that in man respects, an SM2 key *isa* EC key.

Reply:
My proposal is therefore to re-instate the security bits for SM2, and make it exactly the same as for EC.

(this also constitutes a challenge back to whoever specified SM2, to come back with better security bits if they get displeased)

Reply:
And for ec the security strength is 0.5 * order (e.g P224 = 112, P256v = 128)

Reply:
@slontis, that calculation is already in place, with commentary

Reply:
It seems this discussion still needs resolution in the code